providers/upgrade/defUpgradeQueue.js

```diff
@@ -0,0 +1,43 @@
+// (c) Copyright 2024, SAP SE and ClearlyDefined contributors. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const { DefinitionVersionChecker } = require('./defVersionCheck')
+const EntityCoordinates = require('../../lib/entityCoordinates')
+const { setup } = require('./process')
+
+class DefinitionQueueUpgrader extends DefinitionVersionChecker {
+  async validate(definition) {
+    if (!definition) return
+    const result = await super.validate(definition)
+    if (result) return result
+
+    await this._queueUpgrade(definition)
+    return definition
+  }
+
+  async _queueUpgrade(definition) {
+    if (!this._upgrade) throw new Error('Upgrade queue is not set')
+    const message = this._constructMessage(definition)
+    await this._upgrade.queue(JSON.stringify(message))
+    this.logger.debug('Queued for definition upgrade ', {
```

Comment:
I've heard this referred to as `recompute` which I see in the code in other places. I don't see `upgrade` in other parts of the code.  Is there a reason to name the queue and method `upgrade` instead of `recompute`.

Reply:
This recompute happens only when schema version is update/upgraded. I am open to renaming it if it improves the readability.